arangod/Pregel/Worker/Worker.cpp

```diff
@@ -157,7 +158,8 @@ void Worker<V, E, M>::_initializeMessageCaches() {
       auto incoming = std::make_unique<ArrayInCache<M>>(std::set<PregelShard>{},
                                                         _messageFormat.get());
       _inCaches.push_back(incoming.get());
-      _outCaches.push_back(new ArrayOutCache<M>(_config, _messageFormat.get()));
+      _outCaches.push_back(new ArrayOutCache<M>(
+          _config, _config->localPregelShardIDs(), _messageFormat.get()));
```

Comment:
Minor: You could also use `_config->localPregelShardIDs()` in the constructor of ArrayOutCache and only pass 2 parameters instead of 3. 

Just ignore my comment in case you want to just use it to be more similar to the "ArrayInCache" constructor api.

Reply:
It is the point of this PR to make `OutCache` more explicitly depend on things that are used from the "config"; in a further step in the coming days we will remove the dependency on `_config` entirely. 

I can hold off on merging this one until this is sorted, maybe, as to make this clear?

Reply:
I don't mind. In case we have a ticket for this and do not forget - just merge it :)